### PR TITLE
use yaml.safe_load instead of yaml.load

### PR DESCRIPTION
**Pull request by ashwini-orchestral**
It is not safe to call yaml.load with any data received from an untrusted source. It allows instantiation of arbitrary objects.  The function yaml.safe_load limits this ability to simple Python objects like integers or lists. For security reasons, yaml.safe_load is used.

### CODE REVIEW

**Review by m4dcoder — CHANGES_REQUESTED** [review]
@ashwini-orchestral Looks like you included changes from another PR/commit into this PR. It is hard to tell what changes relate to this PR. Can you clean this up? You should branch from master and then apply the changes.

File: st2common/st2common/util/spec_loader.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def load_spec(module_name, spec_file, allow_duplicate_keys=False):
 
     # 1. Check for duplicate keys
     if not allow_duplicate_keys:
-        yaml.load(spec_string, UniqueKeyLoader)
```
Comment (Kami) [inline st2common/st2common/util/spec_loader.py:80]: This function is only used to load ``openapi.yaml.j2`` file which is controlled by us (developers), which means there should be no chance for "yaml type injection", but I do agree and still think it's better and safer to use ``yaml.safe_load``.

**Comment by Kami** [comment]
@ashwini-orchestral As @m4dcoder said, please remove unrelated changes so we can review ``yaml.safe_load`` change in isolation.

**Comment by Kami** [comment]
Superseded by #5162.

**Pull request closed without merging**